src/Collections-Abstract/Collection.class.st

```diff
@@ -340,23 +340,30 @@ Collection >> asSortedCollection: aSortBlock [
 
 { #category : #printing }
 Collection >> asStringOn: aStream delimiter: delimString [
+
 	"Print elements on a stream separated
 	with a delimiter String like: 'a, b, c'
 	Uses #asString instead of #print:."
 
-	self do: [:elem | aStream nextPutAll: elem asString]
-		separatedBy: [aStream nextPutAll: delimString]
+	"String streamContents: [:s| 'abcd' asStringOn: s delimiter: '->'] >>> 'a->b->c->d'"
```

Comment:
May be this is too early but you should pay attention that >>> is not sugar but a real binary message so we should put parentheses around String ...] so hat the >>> is sent to the resulting result 

I think that we should do that all over the comments. 
Normally we should have an extension of DrTests that is check comments.

Reply:
I tried to add the parenthesis, but the current implementation of executable comments seems to split the line on the first >>> then parse and evaluate the left side and the right side independently. Precedence and anything is therefore just ignored. So, missing parenthesis as not signaled when I run the tests.

Fun example: `(1 >>> 1)` is considered a syntactic error because unmatched `(` in the first part